File: web/src/db/normalize.ts
// Verbatim port of `medlens.artifacts.build_normalization.normalize_lookup_text`.
// Casefold, replace non-alphanumerics with spaces, collapse whitespace.
// Used for exact alias lookup against `drug_alias.normalized_alias`.

const NON_ALNUM_RE = /[^a-z0-9]+/g;

export function normalizeLookupText(value: string): string {
  const folded = (value ?? "").toLowerCase().trim();
  const stripped = folded.replace(NON_ALNUM_RE, " ");
  return stripped.split(/\s+/).filter(Boolean).join(" ");
}

File: android/core-data/src/main/java/com/medlens/core/data/util/Normalize.kt
package com.medlens.core.data.util

private val NON_ALNUM_RE = Regex("[^a-z0-9]+")

fun normalizeLookupText(value: String): String {
    val folded = value.lowercase().trim()
    val stripped = folded.replace(NON_ALNUM_RE, " ")
    return stripped.split(Regex("\\s+")).filter { it.isNotBlank() }.joinToString(" ")
}
